Support Style Dictionary v4 and add preprocessors config

mrajaeim opened the pull request.

nado1001 commented on package.json:
```diff
@@ -50,14 +50,14 @@
     "rimraf": "3.0.2",
     "semantic-release": "21.1.2",
-    "style-dictionary": "3.9.2",
+    "style-dictionary": "^4.1.0",
```
It would be great if you could get it to v4.1.1

nado1001 reviewed (changes requested):
Thank you for your contribution!
I'm sorry, but is it possible to have the code in the `README.md` also rewritten as follows?

``````js
import StyleDictionary from 'style-dictionary'
import { makeSdTailwindConfig } from 'sd-tailwindcss-transformer'

const StyleDictionaryTailwind = new StyleDictionary(
  makeSdTailwindConfig({ type: 'all' })
)
await StyleDictionaryTailwind.buildAllPlatforms()
```

mrajaeim commented:
Hi @nado1001,

Thanks for reminding me to update the README. I’ve updated it and also added an integration test to ensure the package works with style-dictionary.

nado1001 reviewed (approved):
Thanks for writing the test as well!

nado1001 reviewed (changes requested):
Sorry, there was an error in `src/tests/integration.test.ts`. Could you please fix it?

mrajaeim commented:
Hi @nado1001 ,

I'm not sure why we're getting a type error in the test for the following line:
`import StyleDictionary from 'style-dictionary'`
This error occurs when running `yarn typecheck` in CI, but the tests pass successfully when running `yarn test`.
It might be a misconfiguration in the TypeScript settings.
Thanks for your review!

nado1001 commented:
Hi, @mrajaeim 
Thanks for the confirmation!
You are right this seems to be caused by the TypeScript configuration.
As it is difficult to solve the problem immediately at the moment, I am sorry, but could you please delete one of the tests you added?
We will add the test you wrote later after we resolve this issue.

mrajaeim commented:
Hi @nado1001,
I’ve deleted the integration test.

nado1001 reviewed (approved):
@mrajaeim 
Thank you very much!
I'll release it right after the merge.
